### PR TITLE
Port new options auth sample

### PR DESCRIPTION
**Pull request by HaoK**
Port of https://github.com/aspnet/AuthSamples/pull/44

### CODE REVIEW

File: src/AuthSamples/samples/Options.MultiTenant/Options.MultiTenant.csproj [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.0</TargetFramework>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore" Version="$(MicrosoftAspNetCorePackageVersion)" />
+    <PackageReference Include="Microsoft.AspNetCore.Authentication.Google" Version="$(MicrosoftAspNetCoreAuthenticationGooglePackageVersion)" />
```
Comment (Tratcher) [inline src/AuthSamples/samples/Options.MultiTenant/Options.MultiTenant.csproj]: Google?

**Review by Tratcher — COMMENTED** [review]
This is giving more false hope than real help. Until you can use it with a real provider like OAuth, OIDC, or JWT, it's not actually useful.

**Comment by HaoK** [comment]
Its not false hope, but more of a basic building block that is needed to configure options per tenant, I can put a big disclaimer in the readme that additional work will need to be done to actually make the security/auth handlers work.   This is basically options + multitenant (ignoring security stack and above)

**Comment by Tratcher** [comment]
> (ignoring security stack and above)

That's the piece our customers need. If we're not going to provide that then why bother explaining the incomplete building blocks?

**Comment by HaoK** [comment]
Because people have asked about these pieces specifically in the past, that will need to do something similar to this to get their multitenancy to work

File: src/AuthSamples/samples/Options.MultiTenant/Options.MultiTenant.csproj (hunk 1 above)
Comment (HaoK) [reply src/AuthSamples/samples/Options.MultiTenant/Options.MultiTenant.csproj]: @natemcmaster what's the new process to update the dependencies.props for AuthSamples?  We are using the google reference to pull in the dependency right now, we just need the MicrosoftAspNetCoreAuthenticationPackageVersion

Comment (natemcmaster) [reply src/AuthSamples/samples/Options.MultiTenant/Options.MultiTenant.csproj]: Update manually, or wait for @jkotalik who is starting on #4246 for auth samples soon.

**Comment by jkotalik** [comment]
@HaoK once https://github.com/aspnet/AspNetCore/pull/6772 is done you can continue with this PR (which should be merged shortly).

**Comment by Eilon** [comment]
Discussed in triage, we feel this sample isn't worth it because it doesn't do enough to help with real scenarios.

**Comment by pseabury** [comment]
@HaoK was this sample abandoned and never ported (per Eilon's comment above)? I certainly don't see it in the new Aspnet samples repo.

**Pull request closed without merging**